src/main/java/seedu/apollo/command/module/ListModuleWithLessonCommand.java

```diff
@@ -226,4 +209,43 @@ private boolean isExistLessonType(ModuleList allModules, LessonType lessonType)
         return false;
     }
 
+    /**
+     * Add a copy of the specific lesson type information into the timetable.
+     *
+     * @param lessonType Specific lesson type to add.
+     * @return The ArrayList of timetable with lesson types in the module list.
+     */
+    private ArrayList<Timetable> copyLessonTypeIntoTimetable(LessonType lessonType) {
+        ArrayList<Timetable> timetableList = new ArrayList<>(module.getModuleTimetable());
+        ArrayList<Timetable> timetableInModuleList = new ArrayList<>();
+
+        for (Timetable timetable : timetableList) {
+            LessonType checkType = determineLessonType(timetable.getLessonType());
+            assert (checkType != null) : "ListModuleWithLessonCommand: Lesson type should not be null!";
+
+            if (checkType.equals(lessonType)) {
+                timetableInModuleList.add(timetable);
+            }
+
+        }
+        return timetableInModuleList;
+    }
```

Comment:
i think u could put this method directly under copyModuleListData instead of all the way at the bottom for readability, since the other methods are less high? in terms of abstraction